src/Language/DifferentialDatalog/Parse.hs

```diff
@@ -485,8 +485,12 @@ atom is_head = withPos $ do
                           else val
        return $ Atom nopos rname $ maybe val' (\b -> E $ EBinding (p1, p2) b val') binding
 
-anonarg = ("",) <$> expr
-namedarg = (,) <$> (dot *> varIdent) <*> (reservedOp "=" *> expr)
+anonarg = ((makeIdentifierWithPos ""),) <$> expr
+namedarg = (,) <$> (dot *> posIdent) <*> (reservedOp "=" *> expr)
+
+-- Identifier with position
+posIdent :: ParsecT String () Identity IdentifierWithPos
+posIdent = withPos $ makeIdentifierWithPos <$> varIdent
```

Comment:
This should probably be called `posVarIdent`, you may want to create `IdentifierWithPos` for things like relation or type names.